iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#include "iceoryx_posh/internal/roudi_environment/roudi_environment.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::runtime;
+using iox::roudi::RouDiEnvironment;
+class PoshRuntime_test : public Test
+{
+  public:
+    PoshRuntime_test()
+    {
+    }
+
+    virtual ~PoshRuntime_test()
+    {
+    }
+
+    virtual void SetUp(){};
+    virtual void TearDown(){};
+
+    void InterOpWait()
+    {
+        std::this_thread::sleep_for(std::chrono::milliseconds(200));
+    }
+
+    iox::cxx::GenericRAII m_errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [](const iox::Error, const std::function<void()>, const iox::ErrorLevel) { m_errorHandlerCalled = true; });
+
+    RouDiEnvironment m_roudiEnv{iox::RouDiConfig_t().setDefaults()};
+    PoshRuntime* m_receiverRuntime{&iox::runtime::PoshRuntime::getInstance("/receiver")};
+    PoshRuntime* m_senderRuntime{&iox::runtime::PoshRuntime::getInstance("/sender")};
+    MqMessage m_sendBuffer;
+    MqMessage m_receiveBuffer;
+    const iox::cxx::CString100 m_runnableName = iox::cxx::CString100("testRunnable");
+    const iox::cxx::CString100 m_invalidRunnableName = iox::cxx::CString100("invalidRunnable,");
```

Comment:
Shouldn't we be using the fixed string API ? I believe `CString100` is deprecated.

Reply:
It should also not be necessary to repeat the type, just use `{"testRunnable"}` instead of `= iox::cxx::string...`

Also, don't we have a type alias for this? If not, maybe we should create one. This should be done in a separate PR, though.

Reply:
fixed